app/controllers/api/v1/ecommerce/warehouses_controller.rb

```diff
@@ -0,0 +1,10 @@
+class Api::V1::Ecommerce::WarehousesController < ActionController::API
+  def list
+    @warehouses = Item.where(sku: params[:sku]).group(:warehouse).count.sort.reverse
```

Comment:
```suggestion
    @warehouses = Item.where(sku: params[:sku]).group(:warehouse).order(sort: :desc).count
```

Acho que se ordenar pelo banco talvez já venha certinho

Reply:
Mas acho que se vocês colocarem um `has_many :items` em warehouses daria pra fazer `Warehouse.joins(:items).group(:code).order(count_all: :desc).count` que já trataria até o map abaixo. ACHO

Reply:
Olá Henrique, utilizei a sua sugestão de mudança. 

Tentei implementar o código do seu texto no commit 364b058 . Ele ficou bem legal dando um resultado mais ou menos: 
{código:quantidade} porém eu não estava conseguindo lidar com o scopo de itens disponíveis desse modo. Decidi então optar pela primeira opção. Mas a segunda me pareceu com potencial. Vou deixar anotado aqui!